client/header/activity-panel/panels/inbox/card.js

```diff
@@ -247,22 +254,34 @@ class InboxNoteCard extends Component {
 
 	renderActions( note ) {
 		const { actions: noteActions, id: noteId } = note;
+		const { clickedActionText } = this.state;
+
 		if ( ! noteActions ) {
 			return;
 		}
+
 		return (
 			<Fragment>
-				{ noteActions.map( ( action, index ) => (
-					<NoteAction
-						key={ index }
-						noteId={ noteId }
-						action={ action }
-					/>
-				) ) }
+				{ clickedActionText !== '' && clickedActionText }
+				{ clickedActionText === '' &&
+					noteActions.map( ( action, index ) => (
+						<NoteAction
+							key={ index }
+							noteId={ noteId }
+							action={ action }
+							onClick={ () => this.onActionClicked( action ) }
+						/>
+					) ) }
 			</Fragment>
 		);
 	}
 
+	onActionClicked = ( action ) => {
+		this.setState( {
```

Comment:
Maybe we can return early here if no action text exists:

```js
if ( ! action.actioned_text ) {
	return;
}
```

Reply:
Done in 3a6052a4